bot/seasons/christmas/adventofcode.py

```diff
@@ -126,7 +128,7 @@ def __init__(self, bot: commands.Bot):
         self.status_task = asyncio.ensure_future(self.bot.loop.create_task(status_coro))
 
     @commands.group(name="adventofcode", aliases=("aoc",), invoke_without_command=True)
-    @override_in_channel()
+    @override_in_channel(AOC_WHITELIST)
```

Comment:
didn't we discover that overriding groups didn't work? isn't that why you're overriding individual commands in the hacktoberfest cog instead?

Reply:
the decorator only works for individual commands with how it is now, I've only added it in on every hacktober command because it's an out of scope hotfix and the AoC commands aren't available.
I was planning to do a PR that would change the decorator to modify the cog class instead of the commands coros after this

Reply:
I think maybe it's overengineering to change the decorator to have it modify the cog class, besides, we'd lose precision. I mean, we still want a command decorator in case we only want one or two commands in a cog to be available, or perhaps that different commands in the same cog need different whitelists.

Ultimately I kind of think we should just apply the current decorator to all the commands in the AoC cog as well and then leave it like that.

Reply:
Fair enough, just thought it'd help with the repeating decorators when it isn't used individually anywhere yet. I'll look if there's a good way to do it by groups but don't have much hope in that

Reply:
I'm not that bothered by a few extra decorators, to be honest. one could argue that this has better readability anyway, since it's more explicit.

Reply:
..But shouldn't this line now be removed, since you've added the decorators to the individual commands instead?

Reply:
it's still a command coro since it has `invoke_without_command`, although the help it provides may not be needed to be allowed in the aoc channel

Reply:
oh yeah that's true. no, then it's fine.